### PR TITLE
Add command LeaderfCommand

### PR DESCRIPTION
**Pull request by tamago324**
I tried to implement `command` which is one of #451.

When executing a command, if a command requires an argument (`E471`), it can now be automatically entered on the command line.

Also, the line can be edited with `e` (normal mode) or `C-o` (input mode) like cmdHistory/searchHistory.

Please let us know if any changes are missing.

Thank you.

### CODE REVIEW

File: autoload/leaderf/python/leaderf/commandExpl.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        lfCmd("let @x = tmp")
+        lfCmd("redir END")
+
+        for line in result.splitlines()[2:]:
+            match = RE_USER_DEFINED_COMMAND.match(line)
+            if match:
+                result_list.append(match.group(1))
+
+        # built-in Ex commands
+        index_file = lfEval("globpath(&rtp, '*/index.txt')")
```
Comment (Yggdroot) [inline autoload/leaderf/python/leaderf/commandExpl.py]: Is `index.txt` denoted to `$VIMRUNTIME/doc/index.txt`? If yes, I don't think we need to search `index.txt` in all the directories of &rtp. We can use  `$VIMRUNTIME/doc/index.txt` directly.

File: autoload/leaderf/python/leaderf/commandExpl.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        # user-defined Ex commands
+        lfCmd("let tmp = @x")
+        lfCmd("redir @x")
+        lfCmd("silent command")
+
+        result = lfEval("@x")
+
+        lfCmd("let @x = tmp")
+        lfCmd("redir END")
```
Comment (Yggdroot) [inline autoload/leaderf/python/leaderf/commandExpl.py]: It's better to use execute() instead.
e.g.,
`result = lfEval("execute('command')")`

File: autoload/leaderf/python/leaderf/commandExpl.py (hunk 1 above)
Comment (tamago324) [reply autoload/leaderf/python/leaderf/commandExpl.py]: > Is index.txt denoted to $VIMRUNTIME/doc/index.txt?

Yes.
It is better to specify directly.
I fixed.

File: autoload/leaderf/python/leaderf/commandExpl.py (hunk 2 above)
Comment (tamago324) [reply autoload/leaderf/python/leaderf/commandExpl.py]: That looks better.
Thanks.

I wrote the code referring to [historyExpl.py](https://github.com/Yggdroot/LeaderF/blob/b339f54e8cb0022557f5cc8e26a43f05aa06b37f/autoload/leaderf/python/leaderf/historyExpl.py#L24-L39).
Should `historyExpl.py` be changed to code that uses `lfEval()` as well?

Comment (Yggdroot) [reply autoload/leaderf/python/leaderf/commandExpl.py]: Let's keep it as it is.

Comment (tamago324) [reply autoload/leaderf/python/leaderf/commandExpl.py]: OK

**Comment by tamago324** [comment]
Thank you!

**Comment by bstaint** [comment]
E492: Not an editor command: LeaderfCommand

**Comment by tamago324** [comment]
> E492: Not an editor command: LeaderfCommand

Thanks for letting me know.

**Pull request merged**